xcp_d/utils/dcan2fmriprep.py

```diff
@@ -11,315 +12,411 @@
-
-        whiteR = glob.glob(anat_dirx + "/fsaverage_LR32k/*R.white.32k_fs_LR.surf.gii")[0]
-        whiteL = glob.glob(anat_dirx + "/fsaverage_LR32k/*L.white.32k_fs_LR.surf.gii")[0]
-
-        dcanimages = [
-            tw1,
-            segm,
-            ribbon,
-            brainmask,
-            tw1,
-            tw1,
-            midL,
-            midR,
-            pialL,
-            pialR,
-            whiteL,
-            whiteR,
-            infL,
-            infR,
-        ]
+    assert isinstance(in_dir, str)
+    assert os.path.isdir(in_dir)
+    assert isinstance(out_dir, str)
+    assert isinstance(sub_id, str)
+
+    subject_dir_fmriprep = os.path.join(out_dir, sub_id)
+
+    # get session ids
+    session_folders = sorted(glob.glob(os.path.join(in_dir, sub_id, "s*")))
+    session_folders = [
+        os.path.basename(ses_dir) for ses_dir in session_folders if os.path.isdir(ses_dir)
+    ]
+    # NOTE: Why split ses- out if you add it right back in?
+    session_ids = [ses_dir.split("ses-")[1] for ses_dir in session_folders]
+    session_ids = [f"ses-{session_id}" for session_id in session_ids]
+
+    # A dictionary of mappings from HCP derivatives to fMRIPrep derivatives.
+    # Values will be lists, to allow one-to-many mappings.
+    copy_dictionary = {}
+
+    for ses_id in session_ids:
+        session_dir_fmriprep = os.path.join(subject_dir_fmriprep, ses_id)
+
+        anat_dir_orig = os.path.join(in_dir, sub_id, ses_id, "files", "MNINonLinear")
+        anat_dir_fmriprep = os.path.join(session_dir_fmriprep, "anat")
+        os.makedirs(anat_dir_fmriprep, exist_ok=True)
+
+        # NOTE: Why *was* this set to the *first* session only? (I fixed it)
+        # AFAICT, this would copy the first session's files from DCAN into *every*
+        # session of the output directory.
+        func_dir_orig = os.path.join(anat_dir_orig, "Results")
+        func_dir_fmriprep = os.path.join(session_dir_fmriprep, "func")
+        os.makedirs(func_dir_fmriprep, exist_ok=True)
+
+        # Collect anatomical files to copy
+        t1w_orig = os.path.join(anat_dir_orig, "T1w.nii.gz")
+        t1w_fmriprep = os.path.join(
+            anat_dir_fmriprep, f"{sub_id}_{ses_id}_desc-preproc_T1w.nii.gz"
+        )
+        copy_dictionary[t1w_orig] = [t1w_fmriprep]
 
-        t1wim = anatdir + sub_id + "_" + sess + "_desc-preproc_T1w.nii.gz"
-        t1seg = anatdir + sub_id + "_" + sess + "_dseg.nii.gz"
-        t1ribbon = anatdir + sub_id + "_" + sess + "_desc-ribbon.nii.gz"
-        t1brainm = anatdir + sub_id + "_" + sess + "_desc-brain_mask.nii.gz"
-        regfile1 = (
-            anatdir + sub_id + "_" + sess + "_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5"
+        # NOTE: We're using the T1w image as a transform. This doesn't make sense.
+        t1w_to_template_fmriprep = os.path.join(
+            anat_dir_fmriprep,
+            f"{sub_id}_{ses_id}_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5",
```

Comment:
The transforms are available in the `MNINonLinear/xfms/` folder.

This transform is named `ANTS_CombinedWarp.nii.gz`, which means we need to support NIFTI transforms in xcp-d.